pkg/controllers/config/configsync.go

```diff
@@ -206,13 +192,19 @@ func translateConfigMap(cloudConfig *v1.ConfigMap, isMultiAZDeployment bool) (*v
 		return nil, fmt.Errorf("failed to modify the provided configuration: %w", err)
 	}
 
+	enableTopologyValue := "false"
+	if enableTopologyFeature {
```

Comment:
IIUC, this enables the topology feature based on the environment, and does not make it configurable as the PR title claims. We should allow the user-provided config to take precedence like we did for `ignore-volume-az`. Possibly a new key under the `openshift-config/cloud-provider-config` CM?
We'll then need to check that updating the CM restarts the pod with the new config as expected.

Reply:
Done